ui/components/nance/DePrize.tsx

```diff
@@ -0,0 +1,224 @@
+import CompetitorABI from 'const/abis/Competitor.json'
+import ERC20 from 'const/abis/ERC20.json'
+import REVDeployer from 'const/abis/REVDeployer.json'
+import {
+  DEPRIZE_DISTRIBUTION_TABLE_ADDRESSES,
+  DEPRIZE_ID,
+  PRIZE_TOKEN_ADDRESSES,
+  COMPETITOR_TABLE_ADDRESSES,
+  PRIZE_DECIMALS,
+  REVNET_ADDRESSES,
+  PRIZE_REVNET_ID,
+} from 'const/config'
+import { TEAM_ADDRESSES } from 'const/config'
+import { useState } from 'react'
+import toast from 'react-hot-toast'
+import { useTeamWearer } from '@/lib/hats/useTeamWearer'
+import toastStyle from '@/lib/marketplace/marketplace-utils/toastConfig'
+import useWindowSize from '@/lib/team/use-window-size'
+import useTokenSupply from '@/lib/tokens/hooks/useTokenSupply'
+import useWatchTokenBalance from '@/lib/tokens/hooks/useWatchTokenBalance'
+import Asset from '@/components/dashboard/treasury/balance/Asset'
+import Container from '@/components/layout/Container'
+import ContentLayout from '@/components/layout/ContentLayout'
+import Head from '@/components/layout/Head'
+import { NoticeFooter } from '@/components/layout/NoticeFooter'
+import { CompetitorPreview } from '@/components/nance/CompetitorPreview'
+import { JoinDePrizeModal } from '@/components/nance/JoinDePrizeModal'
+import StandardButton from '../layout/StandardButton'
+
+export type Metadata = {}
+export type Competitor = {
+  id: string
+  deprize: number
+  teamId: number
+  metadata: Metadata
+}
+export type Distribution = {
+  deprize: number
+  address: string
+  timestamp: number
+  distribution: { [key: string]: number }
+}
+
+export type DePrizeProps = {
+  competitors: Competitor[]
+  distributions: Distribution[]
+  refreshRewards: () => void
+}
+
+export function DePrize({
+  competitors,
+  distributions,
+  refreshRewards,
+}: DePrizeProps) {
+  const chain = process.env.NEXT_PUBLIC_CHAIN === 'mainnet' ? Arbitrum : Sepolia
+  const { isMobile } = useWindowSize()
+  const [joinModalOpen, setJoinModalOpen] = useState(false)
+  const userAddress = useAddress()
+  const quarter = Math.floor((new Date().getMonth() + 3) / 3) - 1
+
+  const { contract: prizeContract } = useContract(
+    PRIZE_TOKEN_ADDRESSES[chain.slug],
+    ERC20.abi
+  )
+  const { contract: competitorContract } = useContract(
+    COMPETITOR_TABLE_ADDRESSES[chain.slug],
+    CompetitorABI
+  )
+  const { contract: teamContract } = useContract(TEAM_ADDRESSES[chain.slug])
+
+  const prizeBalance = useWatchTokenBalance(prizeContract, PRIZE_DECIMALS)
+  const userHasVotingPower = prizeBalance > 0
+  const prizeSupply = useTokenSupply(prizeContract, PRIZE_DECIMALS)
+  const prizeBudget = prizeSupply * 0.1
+  const winnerPool = prizeSupply * 0.3
+  const prizePrice = 1
+
+  const userTeams = useTeamWearer(teamContract, chain, userAddress)
+
+  const isCompetitor = userTeams.some((team: any) =>
+    competitors.some(
+      (competitor) => competitor.teamId.toString() === team.teamId
+    )
+  )
+  const handleJoinWithTeam = async (teamId: string) => {
```

Comment:
So I guess it looks like:

- Apply to become a team (approval process)
- If you have a team you can join permissionlessly to a prize

I think that works (v.s. apply for each prize). As long as maybe the prize operator can remove a competitor if there's something weird that happened. But I think the optimistic model makes sense.